### ISSUE TITLE
Lightning Protocol Tests

### ISSUE THREAD

**Issue by TheBlueMatt** [rust-bitcoin/rust-lightning#406]
Rusty's been working on a protocol test framework including a protocol test language that we'd want to implement a parser for. We should probably play with doing it before we end up with a language that doesn't map well to our API.

Current branch is at https://github.com/ElementsProject/lightning-rfc-protocol-test

**Comment by jkczyz** [comment]
Writing a parser shouldn't be necessary for using the protocol test framework. Rather, we'd write a runner that implements the [interface](https://github.com/ElementsProject/lightning-rfc-protocol-test/blob/203a02598d2a7e2a2db03972c89b7f22c6aef58f/tools/test-events.py#L189) defined by the framework similar to [c-lightning's runner](https://github.com/ElementsProject/lightning-rfc-protocol-test/blob/203a02598d2a7e2a2db03972c89b7f22c6aef58f/tools/test-events-clightning.py#L116).

Essentially, the runner acts as an adapter from the test framework interface to our own API. Do you envision using the framework in a different manner?

**Comment by TheBlueMatt** [comment]
I hadn't looked into it at all, feel free to do it any way you feel like if you're gonna tackle it. It may be more effort to add rust-lightning bindings in python to just use their python framework than write a parser for the events files at https://github.com/ElementsProject/lightning-rfc-protocol-test/tree/master/tests/events.

**Comment by jkczyz** [comment]
My main objection to writing a separate parser (not to mention a test framework) is that it adds an unnecessary maintenance burden.

Thinking about it more, this could be avoided instead by writing a rust binary implementing barebones functionality (similar to what we've discussed before). The runner then would call into this in some manner (e.g., RPC interface or similar).

**Comment by TheBlueMatt** [comment]
Right, fair enough. There’s a simple demo that uses Bitcoin Core RPC calls at https://github.com/TheBlueMatt/rust-lightning-bitcoinrpc which may be a reasonable starting point.
